### PR TITLE
[mono] Fix 32bit Android build

### PR DESCRIPTION
**Pull request by EgorBo**
1) Use `HostArch` for AppBuilder tasks 
2) Remove `tkill` for Android 32bit since it's not available in recent Android NDKs anymore.

### CODE REVIEW

File: src/mono/msbuild/AndroidAppBuilder/Templates/runtime-android.c [hunk 1]
```diff
@@ -184,6 +185,12 @@ strncpy_str (JNIEnv *env, char *buff, jstring str, int nbuff)
         (*env)->ReleaseStringUTFChars (env, str, copy_buff);
 }
 
+int
+tkill(pid_t tid, int sig)
```
Comment (akoeplinger) [inline src/mono/msbuild/AndroidAppBuilder/Templates/runtime-android.c:189]: shouldn't this be in the runtime somewhere?

Comment (EgorBo) [reply src/mono/msbuild/AndroidAppBuilder/Templates/runtime-android.c:189]: hm.. it shouldn't be needed at all for Android - it was removed from NDK

Comment (akoeplinger) [reply src/mono/msbuild/AndroidAppBuilder/Templates/runtime-android.c:189]: then I'm even more confused what this does :D

Comment (EgorBo) [reply src/mono/msbuild/AndroidAppBuilder/Templates/runtime-android.c]: removed it so now we rely on `pthread_kill` on Android 32bit. Tests seem pass

**Review by steveisok — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**